Feature/add linters

DZariusz opened the pull request.

DZariusz committed:
chore(*): add linters and fix js eslint errors

Include linters for js and solidity. Fix all linters errors for javascript.
Add exception rule for `prefer-destructuring` for array, so we can use i.e: var foo = array[0];

enddynayn commented:
👍

enddynayn commented:
nice! thank you.

avoloshko commented on contracts/Plasma.sol:
```diff
@@ -153,55 +155,57 @@ contract Plasma is Ownable {
   // challenge
   function challengeWithdrawDeposit(uint _blockIndex, bytes _transactionBytes, bytes _proof, bytes signature) public returns (bool success) {
-    Block memory block = blocks[_blockIndex];
+    Block memory newBlock = blocks[_blockIndex];
```
it's not new. it's a block where the deposit has been spent.

avoloshko commented on contracts/Plasma.sol:
```diff
@@ -153,55 +155,57 @@ contract Plasma is Ownable {
-  function proveTX(uint _blockIndex, bytes _transactionBytes, bytes _proof) public constant returns (bool success) {
-    Block memory block = blocks[_blockIndex];
+  function proveTX(uint _blockIndex, bytes _transactionBytes, bytes _proof) public view returns (bool success) {
+    Block memory newBlock = blocks[_blockIndex];
```
why 'new'?

DZariusz replied:
because `block` is global variable in solidity and compiler give us a warning for it.

therevoltingx commented:
@DZariusz 
Let's pair up and fix the issues with codeship and get this merged!

DZariusz committed:
chore(*): fix solidity compiler and linter errors

Fix solidity compile warnings.
Fix solidity linters errors - but not all. There are still 8 compiler errors regarding
inline assembly. There are still 26 linter errors (most of them about assembly, block.timestamp,
the rese about: avoid assigning to function parameters). I set them to be ignored,
because at this point idk, if out implementation will allow us to not use
features that causes this linters errors.
